crates/bevy_reflect/src/type_registry.rs

```diff
@@ -599,6 +602,39 @@ impl<T: Reflect> FromType<T> for ReflectFromPtr {
     }
 }
 
+/// Associate a `TraitTypeData`
+pub trait TraitTypeDataRelevance {
+    type TypeData: TraitTypeData;
+}
```

Comment:
This probably needs more documentation considering manual implementors may want to make use of it.